### DISCUSSION TITLE
bref console commands error => Fatal error: Uncaught TypeError: base64_decode() expects parameter 1 to be string, null given

### DISCUSSION THREAD

**Discussion by Houssem-Esprit** [brefphp/bref#968] [Support] [answered]
So, I'm about to create a serverless api platform with aws lambda using bref. it goas well with the deployment on lambda and connection to MariaDB RDS, then when I jumped to the database operations within my serverless api platform using [bref console commands](https://bref.sh/docs/runtimes/console.html) I kept stuck with the same error `Fatal error: Uncaught TypeError: base64_decode() expects parameter 1 to be string, null given`, the command I used it is `vendor/bin/bref --region=eu-west-1 --profile=default cli symfony-prod-console -- doctrine:database:create`. I did try a bunch of combinations with [options] and without, environment variables, and still can't figure out why this problem came in front.

---------
>Running command: $ vendor/bin/bref --region=eu-west-1 --profile=default cli symfony-prod-console -- doctrine:database:create -vvv  
stack trace : 
`
Fatal error: Uncaught TypeError: base64_decode() expects parameter 1 to be string, null given in C:\Users\HP\Desktop\2020\Confledis\Projects\Nx_Demo\api\backend-module-federation\vendor\bref\bref\src\Lambda\InvocationResult.php:29
Stack trace:
#0 C:\Users\HP\Desktop\2020\Confledis\Projects\Nx_Demo\api\backend-module-federation\vendor\bref\bref\src\Lambda\InvocationResult.php(29): base64_decode(NULL)
#1 C:\Users\HP\Desktop\2020\Confledis\Projects\Nx_Demo\api\backend-module-federation\vendor\bref\bref\src\Lambda\InvocationFailed.php(20): Bref\Lambda\InvocationResult->getLogs()
#2 C:\Users\HP\Desktop\2020\Confledis\Projects\Nx_Demo\api\backend-module-federation\vendor\bin\bref(21) : eval()'d code(109): Bref\Lambda\InvocationFailed->getInvocationLogs()
#3 [internal function]: Silly\Application->{closure}('symfony-prod-co...', 'eu-west-1', 'default', Array, Object(Symfony\Component\Console\Style\SymfonyStyle))
#4 C:\Users\HP\Desktop\2020\Confledis\Projects\Nx_Demo\api\backend-module-federation\vendor\php-di\invoker\src\Invoker in C:\Users\HP\Desktop\2020\Confledis\Projects\Nx_Demo\api\backend-module-federation\vendor\bref\bref\src\Lambda\InvocationResult.php on line 29
` 
---------
 I'm using : 
- PHP 7.3
- Symfony 5.3
- Api Platform 2.6
-  bref 1.2.8
- symfony-bridge : 0.1.2
- Windows 7

can anybody tell me what I'm missing here, or if by any chance this is a problem with bref console commands?

**Comment by t-richard** [comment]
@Houssem-Esprit do you have a complete stack trace of the error ? This error by itself is too broad to get something out of it. Maybe try to run it with verbose or debug enabled to get more info. Maybe there is something more useful in Cloudwatch ?

Also, have you specified you server version to doctrine in the `DATABASE_URL` to match the one used by RDS ? This can cause weird issues like this.

Lastly, you can tell RDS to create the database for you when creating it, its hidden in `Additional configuration > Initial database name` in the creation form. It's also configurable if using Cloudformation.

**Reply by Houssem-Esprit** [reply]
@t-richard  I updated the post above with the stack trace of the bref console commands error

**Comment by Houssem-Esprit** [comment, marked as answer]
After looking to CloudWatch logs for console function, I figured out the problem I have. for instant the error output of the "beef console commands" looks so meaningless, it should change.

**Reply by t-richard** [reply]
@Houssem-Esprit Glad you found out.

Could you say a bit more about what the problem was ? Maybe it's actionnable and we can find a fix for it ?

**Reply by Houssem-Esprit** [reply]
The problem for itself is silly, I run the command to create a database while I already have one, so the cloudwatch stack was `General error: 1007 Can't create database 'dbname'; database exists`.
with another command like "make:migration'' the problem is that symfony/make bundle is limited to execute only on dev environment. what annoying here is the same output error (almost) on terminal for all failed commands. For example, why isn't the CloudWatch stack trace delivered as an output error since the error isn't produced by a miss-formed command, but was produced on an aws service.

**Reply by t-richard** [reply]
Actually it should be doing just that : getting the logs associated to you Lambda invocation from Cloudwatch. The error you have is `base64_decode() expects parameter 1 to be string, null given in [...] bref\src\Lambda\InvocationResult.php:29`

This is the actual code there https://github.com/brefphp/bref/blob/master/src/Lambda/InvocationResult.php#L27-L30

It tries to get the logs from the Lambda invocation. But it gets `null` so if logs are present, somehow the SDK is not able to retrieve them. Maybe it's an IAM permission problem ? Does your AWS profile has access to Cloudwatch logs ?

I can't reproduce with limited permissions though.
